### PR TITLE
Fail the tool when analyzers fail to load

### PR DESCRIPTION
**Pull request by Smaug123**
The tool currently silently swallows this sort of error.

This change is backward incompatible. Is that OK?

### CODE REVIEW

File: src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.Client.fs [hunk 1]
```diff
@@ -185,6 +194,8 @@ type Client<'TAttribute, 'TContext when 'TAttribute :> AnalyzerAttribute and 'TC
                     then
                         true
                     else
+                        System.Threading.Interlocked.Increment skippedAssemblies |> ignore
```
Comment (Smaug123) [inline src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.Client.fs:197]: (This is an idiom I strongly prefer, because it's robust against people choosing in the future to parallelise the `Array.filter`. Of course, I could rewrite this as a fold instead if mutability is considered harmful.)

File: src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.Client.fs [hunk 2]
```diff
@@ -233,10 +244,22 @@ type Client<'TAttribute, 'TContext when 'TAttribute :> AnalyzerAttribute and 'TC
                 registeredAnalyzers.AddOrUpdate(path, analyzers, (fun _ _ -> analyzers))
                 |> ignore
 
-            Array.length analyzers, analyzers |> Seq.collect snd |> Seq.length
+            let assemblyCount = Array.length analyzers
+            let analyzerCount = analyzers |> Seq.sumBy (snd >> Seq.length)
```
Comment (Smaug123) [inline src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.Client.fs:248]: Drive-by efficiency change

**Review by dawedawe — APPROVED** [review]
Thanks, makes sense.

**Comment by dawedawe** [comment]
@Smaug123 One thing: Could you add a changelog entry, please.

**Review by nojaf — APPROVED** [review]
Hi, I'm also on board with this.

The changelog entry (in [CHANGELOG.md](https://github.com/ionide/FSharp.Analyzers.SDK/blob/main/CHANGELOG.md)) would look like:

```
## [Unreleased]

### Changed
* [Good entry here](https://github.com/ionide/FSharp.Analyzers.SDK/pull/198) (thanks @Smaug123)
```

Out of interest, what assemblies failed to load in your case?
Actual assemblies with analyzers or other DLLs it tried to load?

**Comment by Smaug123** [comment]
This was motivated by https://github.com/Smaug123/fsharp-prattparser and specifically https://github.com/Smaug123/fsharp-prattparser/actions/runs/7618965411/job/20751163083 :

```
info: Running in verbose mode
info: Treat as Hints: []
info: Treat as Info: []
info: Treat as Warning: []
info: Treat as Error: [GRA-IMMUTABLECOLLECTIONEQUALITY-001, GRA-INTERPOLATED-001, GRA-JSONOPTS-001, GRA-LOGARGFUNCFULLAPP-001, GRA-STRING-001, GRA-STRING-002, GRA-STRING-003, GRA-TYPE-ANNOTATE-001, GRA-UNIONCASE-001, GRA-VIRTUALCALL-001]
info: Exclude Files: []
info: Include Files: []
info: Loading analyzers from /home/runner/work/fsharp-prattparser/fsharp-prattparser/.analyzerpackages/g-research.fsharp.analyzers/0.6.0/
error: Trying to load /home/runner/work/fsharp-prattparser/fsharp-prattparser/.analyzerpackages/g-research.fsharp.analyzers/0.6.0/analyzers/dotnet/fs/G-Research.FSharp.Analyzers.dll which was built using SDK version 0.22.0.0. Expect 0.23.0.0 instead. Assembly will be skipped.
info: Registered 0 analyzers from 0 dlls
info: No messages found from the analyzer(s)
```
